user-access-handlers/src/main/java/no/unit/nva/handlers/IdentityServiceInitHandler.java

```diff
@@ -92,50 +82,14 @@ private void logError(Exception exception) {
         logger.warn(exception.getMessage());
     }
     
-    private RoleDto addRole(RoleDto role) throws ConflictException, InvalidInputException {
-        logger.info("Adding role:{}", role);
-        identityService.addRole(role);
+    private RoleDto addOrUpdateRole(RoleDto role) throws InvalidInputException, NotFoundException {
+        try {
+            identityService.addRole(role);
+            logger.info("Added role: {}", role);
+        } catch (ConflictException e) {
+            identityService.updateRole(role);
+            logger.info("Updated role: {}", role);
+        }
         return role;
     }
-    
-    private List<RoleDto> createDefaultRoles() {
```

Comment:
Better to have this in own class as you have done. My opinion is to keep as minimal code as possible in the handler itself, and rather delegate to other classes per purpose

Reply:
Agreed